modules/cloudsql-instance/variables.tf

```diff
@@ -250,8 +250,11 @@ variable "replicas" {
 
 variable "root_password" {
```

Comment:
nit: you could add a validation to ensure only one of password or random_password is provided.

Something like this (untested):
```
validation {
  condition = (var.root_password.password != null) != var.root_password.random_password
  error_message = "Cannot provide root_password.password and root_password.random_password at the same time"
}
```

Reply:
It's valid to provide password=null and random_password=false (this should pass null to the provider - current behaviour). I think the only invalid situation is password != null and random_password = true.

Reply:
I used the following, because it seemed clearer than negating the terms in the expression. Thoughts?
```
  validation {
    condition     = !(var.root_password.password != null && var.root_password.random_password)
    error_message = "Cannot provide root_password.password and root_password.random_password at the same time"
  }
```